### PR TITLE
Try out composite GH actions

### PR DESCRIPTION
**Pull request by snazy**
This change is [Reviewable](https://reviewable.io/reviews/projectnessie/nessie/1849)

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -20,74 +20,14 @@ jobs:
...
-        sudo dd if=/dev/zero of=/mnt/swapfile-2 bs=1MiB count=$((8*1024))
-        sudo mkswap /mnt/swapfile-2
-        sudo swapon /mnt/swapfile-2
     - uses: actions/checkout@v2
-    - name: Set up JDK 11
-      uses: actions/setup-java@v1
+    - name: Setup runner
+      uses: ./.github/actions/setup-runner
+    - name: Setup Java, Maven, Gradle
+      uses: ./.github/actions/dev-tool-java
```
Comment (rymurr) [inline .github/workflows/main.yml:27]: how are things ordered in the composite actions? Do we have any ordering requirements?

Comment (snazy) [reply .github/workflows/main.yml:27]: Not really. It's more a "it feels more natural to setup the runner first" thing.

Comment (rymurr) [reply .github/workflows/main.yml:27]: ah, no sorry. I meant in the dev-tool-java action. That is a composite action, is the order of runs defined?

Comment (snazy) [reply .github/workflows/main.yml:27]: It's all sequential.

**Review by rymurr — APPROVED** [review]

**Pull request merged**